tests/accessibility/index.test.js

```diff
@@ -0,0 +1,163 @@
+// @ts-check
+const path = require('path');
+const { test } = require('mocha');
+const axe = require('axe-core');
+const jsdom = require('jsdom');
+const fetch = require('node-fetch').default;
+const { A11yError } = require('@sa11y/format');
+const cheerio = require('cheerio');
+
+const config = require('../../lib/config');
+const helperServer = require('../helperServer');
+
+const SITE_URL = 'http://localhost:' + config.serverPort;
+const EXAMPLE_COURSE_DIR = path.resolve(__dirname, '..', '..', 'exampleCourse');
+
+/**
+ * Loads the given URL into a Cheerio object.
+ *
+ * @param {string} url
+ * @returns {Promise<cheerio.Root>}
+ */
+async function loadPageCheerio(url) {
+  const text = await fetch(url).then((res) => {
+    if (!res.ok) {
+      throw new Error(`Error loading page: ${res.status}`);
+    }
+    return res.text();
+  });
+
+  console.log(text);
+  return cheerio.load(text);
+}
+
+/**
+ * Loads the given URL into a JSDOM object.
+ *
+ * @param {string} url
+ * @returns {Promise<import('jsdom').JSDOM>}
+ */
+async function loadPageJsdom(url) {
+  const text = await fetch(url).then((res) => {
+    if (!res.ok) {
+      throw new Error(`Error loading page: ${res.status}`);
+    }
+    return res.text();
+  });
+  return new jsdom.JSDOM(text);
+}
+
+/**
+ * Checks the given URL for accessibility violations.
+ *
+ * @param {string} url
+ */
+async function checkPage(url) {
+  const page = await loadPageJsdom(SITE_URL + url);
+  const results = await axe.run(page.window.document.documentElement);
+  A11yError.checkAndThrow(results.violations);
+  // reportViolations(url, results);
+}
+
+async function getExampleCourseContext() {
+  const $ = await loadPageCheerio(SITE_URL);
+
+  const courseLink = $('td a:contains("XC 101: Example Course")');
+  const courseLinkHref = courseLink.attr('href');
+  const courseIdMatch = courseLinkHref.match(/course\/(\d+)/);
+  if (!courseIdMatch) {
+    throw new Error(`Could not find course ID in link: ${courseLinkHref}`);
+  }
+  const courseId = courseIdMatch[1];
+
+  const courseInstanceLink = courseLink.parents('tr').find('td a:contains("Spring 2015")');
+  const courseInstanceLinkHref = courseInstanceLink.attr('href');
+  const courseInstanceIdMatch = courseInstanceLinkHref.match(/course_instance\/(\d+)/);
+  if (!courseInstanceIdMatch) {
+    throw new Error(`Could not find course instance ID in link: ${courseInstanceLinkHref}`);
+  }
+  const courseInstanceId = courseInstanceIdMatch[1];
+
+  return {
+    courseId,
+    courseInstanceId,
+  };
+}
+
+// These are meant to be a representative sample of pages from across the site.
+// We hardcode database IDs to 1, as such entities will always exist. We don't
+// case exactly which entities are used, just that we get a representative page
+// of each "kind".
+const pages = [
+  '/',
```

Comment:
It would be cool if we could automatically generate this list from `server.js` somehow so that it wasn't an extra thing to remember to update when I'm don't know of any clean way to do that though :sweat_smile:.

Maybe easier to implement: should we also make sure we have an example of every element in here too? We could grab the list from `tests/exampleCourseQuestions.test.js` and/or `tests/testCourseQuestions.test.js`

Reply:
I was thinking about the former a bit, but ultimately I think it may be more trouble than it's worth.

It'd be great to have some elements represented in these tests too!

Reply:
I wonder how far a simple Regex could take us...something like just doing `/'pl/course_instance.*'/` on all of `server.js`. 
Of course as you say, it could be more trouble than its worth.

Reply:
You're welcome to try to figure something out, but I think our time would be better spent remediating accessibility issues than trying to find clever ways to ensure perfect coverage.

Reply:
It turns out this wasn't actually *too* bad to get working! See 252b52c and subsequent commits.

Reply:
Very cool!